analyzers/nodejs/mocks_test.go

```diff
@@ -25,3 +27,17 @@ func (n MockNPM) Clean(dir string) error {
 func (n MockNPM) Install(dir string) error {
 	return nil
 }
+
+type MockNPMFailure struct{}
+
+func (n MockNPMFailure) List(_ string) (npm.Output, error) {
+	return npm.Output{}, errors.New("expected failure")
```

Comment:
I'd prefer if these error messages had different strings so it would be easier to distinguish them.